### PR TITLE
[10.0][FIX]stock_Account

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
If no operating unit in the locations take it from the warehouse.

UPDATE: updated PR for a couple of extra fixes:
- post stock_journal inter-OU entry after creating it
- do not create entry if none of the locations involved has a company

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
forever_alone

File: stock_account_operating_unit/model/stock_quant.py [hunk 1]
```diff
@@ -22,11 +22,11 @@ def _account_entry_move(self, move):
         if move.product_id.valuation == 'real_time':
             # Inter-operating unit moves do not accept to
             # from/to non-internal location
-            if (
+            if ((move.location_id.company_id or
+                 move.location_dest_id.company_id) and
```
Comment (MiquelRForgeFlow) [inline stock_account_operating_unit/model/stock_quant.py]: Redundant comparison. Make it simple:

```python
              if (move.location_id.company_id and
              move.location_id.company_id ==
                    move.location_dest_id.company_id and
                    move.operating_unit_id != move.operating_unit_dest_id):
```

Comment (AaronHForgeFlow) [reply stock_account_operating_unit/model/stock_quant.py]: Ok, It is fine, but it is as complicated as I did :smile:

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by HviorForgeFlow — APPROVED** [review]

**Review by JoanSForgeFlow — APPROVED** [review]

**Pull request merged**